core/scripts/PrecisionErrors.js

```diff
@@ -110,6 +113,15 @@ async function createTestEnvironment() {
   emp = await ExpiringMultiParty.new(constructorParams, { from: contractDeployer });
   synthetic = await SyntheticToken.at(await emp.tokenCurrency());
 
+  // Need to register the contract so that it can add/remove party members, even though
+  // registration is not required to make price requests specifically to the MockOracle
+  // (n.b. it is required to be registered to make price requests to the production Oracle).
+  registry = await Registry.deployed();
```

Comment:
the need to register the contract makes this a bit of a pain in the contexts of the tests/scripts but it has to happen if we want to implement this logic.

Reply:
totally agreed